pkg/serverless/trace/inferredspan/span_enrichment.go

```diff
@@ -217,10 +220,120 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithS3Event(eventPayload eve
 	}
 }
 
+type DatadogPayload struct {
+	Type        string `json:"Type"`
+	Value       string `json:"Value"`
+	DataType    string `json:"dataType"`
+	BinaryValue string `json:"binaryValue"`
+	StringValue string `json:"stringValue"`
+}
+
+type TraceHeader struct {
+	TraceID          string `json:"x-datadog-trace-id"`
+	ParentID         string `json:"x-datadog-parent-id"`
+	SamplingPriority string `json:"x-datadog-sampling-priority"`
+	Sampled          string `json:"x-datadog-sampled"`
+}
+
+type customMessageAttributeStruct struct {
+	Type  string `json:"Type"`
+	Value string `json:"Value"`
+}
+type BodyStruct struct {
+	Message           string                                  `json:"Message"`
+	MessageAttributes map[string]customMessageAttributeStruct `json:"MessageAttributes"`
+}
+type CustomDatadogPayload struct {
+	Type        string `json:"Type"`
+	Value       string `json:"Value"`
+	DataType    string `json:"dataType"`
+	BinaryValue string `json:"binaryValue"`
+	StringValue string `json:"stringValue"`
+}
+
+func extractContextFromSNSSQSEvent(firstRecord events.SQSMessage) (string, string, string, error) {
+	var bodyStruct BodyStruct
+	if err := json.Unmarshal([]byte(firstRecord.Body), &bodyStruct); err != nil {
+		return "", "", "", err
+	}
+	var ddCustomPayloadValue, ok = bodyStruct.MessageAttributes["_datadog"]
+	if !ok {
+		return "", "", "", errors.New("no Datadog trace context found")
+	}
+
+	var traceData TraceHeader
+	if ddCustomPayloadValue.Type == "Binary" {
+		decodedBinary, err := base64.StdEncoding.DecodeString(string(ddCustomPayloadValue.Value))
+		if err != nil {
+			return "", "", "", err
+		}
+		if err := json.Unmarshal(decodedBinary, &traceData); err != nil {
+			return "", "", "", err
+		}
+	} else {
+		return "", "", "", errors.New("unsupported DataType in _datadog payload")
+	}
+
+	return traceData.TraceID, traceData.ParentID, traceData.SamplingPriority, nil
+}
+
+func extractContextFromPureSqsEvent(ddPayloadValue events.SQSMessageAttribute) (string, string, string, error) {
+	var traceData TraceHeader
+	if ddPayloadValue.DataType == "String" {
+		if err := json.Unmarshal([]byte(*ddPayloadValue.StringValue), &traceData); err != nil {
+			return "", "", "", err
+		}
+	} else {
+		return "", "", "", errors.New("unsupported DataType in _datadog payload")
+	}
+
+	return traceData.TraceID, traceData.ParentID, traceData.SamplingPriority, nil
+}
+
+func convertToUint64(str string) (uint64, error) {
+	converted, err := strconv.ParseUint(str, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return converted, nil
+}
+
+func convertToFloat64(str string) (float64, error) {
+	converted, err := strconv.ParseFloat(str, 64)
+	if err != nil {
+		return 0, err
+	}
+	return converted, nil
+}
+
+func convertValues(traceID, parentID, samplingPriority string) (uint64, uint64, float64, error) {
```

Comment:
This method has a few problems;
1. It returns a list of values which are unnamed and very easy to swap. It'd be very easy for someone to mistake the traceID and parentID here. Please convert it to return a struct.
2. The method name is quite vague, I think we could make it more clear, wdyt?